UnityProject/Assets/Scripts/Mobs/BrainAI/States/Arial/LookForTroubleArialAi.cs

```diff
@@ -14,91 +18,76 @@ namespace Mobs.BrainAI.States.Arial
 	public class LookForTroubleArialAi : BrainMobState
 	{
 		private GameObject target;
+		private MobTraversal pathfinder => master.Traversal;
 		[SerializeField] private BrainWanderState wanderState;
 		[SerializeField] private CauseTroubleArialAi troubelState;
-		[SerializeField] private MobPathfinderV2 pathfinder;
 		[SerializeField] private List<AddressableAudioSource> stateEnterSounds = new List<AddressableAudioSource>();
 
-		private List<Node> path = new List<Node>();
+		public List<TraversalStrat> TraversalStrategies = new List<TraversalStrat>();
 
-		public void Start()
+		private bool isTraversing = false;
+
+
+		private void Start()
 		{
-			base.Awake();
-			pathfinder = LivingHealthMaster.playerScript.GetComponent<MobPathfinderV2>();
-			if (pathfinder == null)
-			{
-				Loggy.Error("[LookForTroubleArialAi] - NO PATHFINDER DETECTED. ARIAL WILL BE STUCK.");
-			}
+			TraversalStrategies.Add(new OpenDoor());
+		}
+
+		public override void OnRemovedFromBody(LivingHealthMasterBase livingHealth, GameObject source = null)
+		{
+			UnsubscribeToPathfinderEvents();
+			base.OnRemovedFromBody(livingHealth, source);
 		}
 
 		public override void OnEnterState()
 		{
+			SubscribeToPathfinderEvents();
+			target = null;
+			isTraversing = false;
 			target = DecideTarget();
 			if (target == null)
 			{
 				//enter wander state.
 				master.AddRemoveState(null, wanderState);
 			}
-
 			SoundManager.PlayNetworkedAtPos(stateEnterSounds.PickRandom(),
 				LivingHealthMaster.gameObject.AssumedWorldPosServer());
 		}
 
 		public override void OnExitState()
 		{
+			UnsubscribeToPathfinderEvents();
 			target = null;
-			path = null;
 		}
 
 		public override void OnUpdateTick()
 		{
-			if (LivingHealthMaster.IsSoftCrit || LivingHealthMaster.IsCrit || LivingHealthMaster.IsDead) return;
+			if (IsStillTraversing()) return;
+			if (LivingHealthMaster.IsSoftCrit || LivingHealthMaster.IsCrit || LivingHealthMaster.IsDead)
+			{
+				isTraversing = false;
+				return;
+			}
+
 			if (target == null)
 			{
 				target = DecideTarget();
 				if (target != null)
 				{
 					master.AddRemoveState(wanderState, null);
 				}
-
 				return;
 			}
 
-			if (pathfinder == null) return;
-			path = pathfinder.FindNewPath(
-				MatrixManager.WorldToLocal(LivingHealthMaster.playerScript.playerMove.OfficialPosition,
-					LivingHealthMaster.RegisterTile.Matrix).RoundTo2Int(),
-				target.gameObject.AssumedWorldPosServer().RoundTo2Int());
-			Loggy.Info($"{target} - {target.OrNull()?.AssumedWorldPosServer()}");
-			if (path != null)
+			if (pathfinder.QueueMovementGoal(target.gameObject.TileLocalPosition().To3Int(),
```

Comment:
If the new target position here is the same as the old one, dont generate new path. 
Should only need to generate a new path if the target moves and/or they encounter an un expected obstacle in their path.

(Not sure if these conditions are implemented elsewhere, hard to tell only seeing the diffs)

Reply:
I thought about that when I first added this function, but then I remembered that we might want mobs to go back to their original position after they reached their goal; so I intentionally removed that from the Queue code.